File: 백준 문제 풀이 모음/class 2/1018.py
n, m = map(int, input().split())
board = list()
for i in range(n) :
    board.append(input())

new = []

for i in range(0, n-8+1):   #row
    for j in range(0, m-8+1) : #column

        w_count = 0
        b_count = 0

        for ii in range(i, i+8) :
            for jj in range(j, j+8) :
        
                if (ii + jj) % 2 == 0 :
                    if board[ii][jj] != 'W' :
                        w_count += 1
                    elif board[ii][jj] != 'B' :
                        b_count += 1

                else :
                    if board[ii][jj] != 'W' :
                        b_count += 1
                    elif board[ii][jj] != 'B' :
                        w_count += 1      
        
        new.append(b_count)
        new.append(w_count)

print(min(new))

File: 백준 문제 풀이 모음/kotlin/class 2/1018.kt
import java.util.*


var min = 100000000

val bw = charArrayOf('B','W','B','W','B','W','B','W')
val wb = charArrayOf('W','B','W','B','W','B','W','B')

fun check(x : Int, y : Int, matrix : Array<CharArray>) : Int {

    var bw_count = 0
    var wb_count = 0

//    for (i in x until x + 8) {
//        for (j in y until y + 8) {
//            print(matrix[i][j])
//            print(' ')
//        }
//        println()
//    }

    for (i in x until x + 8) {
        for (j in y until y+8) {
            if (i % 2 == 0) {
                if (!matrix[i][j].equals(bw[j-y])) {
                    bw_count ++
                }

                if (!matrix[i][j].equals(wb[j-y])) {
                    wb_count ++
                }
            } else {
                if (!matrix[i][j].equals(wb[j-y])) {
                    bw_count ++
                }

                if (!matrix[i][j].equals(bw[j-y])) {
                    wb_count ++
                }
            }
        }
    }

//    println(wb_count)
//    println(bw_count)
    return Math.min(wb_count, bw_count)

}


fun main()= with(System.`in`.bufferedReader()) {
    val str = readLine().split(' ').toList()
    val n = str[0].toInt()
    val m = str[1].toInt()
    val matrix = Array(n) { CharArray(m) {'0'} }


    for (i in 0 until n) {
        val input = readLine().toCharArray()
        for ( j in 0 until m) {
            matrix[i][j] = input[j]
        }
    }

    for (i in 0 .. n-8) {
        for (j in 0 .. m-8) {

            min = Math.min(min, check(i, j, matrix))
        }
    }

    println(min)
//    val sb = StringBuilder()
//    matrix.forEach {
//        it.forEach {
//            sb.append(it)
//            sb.append(' ')
//        }
//        sb.append('\n')
//    }
//
//    println(sb.toString())


}
